File: Vishwas RNO 40/DotNet/Assignment 6/TourLib/PremiumTours.cs
namespace Tours;

public class PremiumTours
{
    [LuxuryTax(10)]
    public double Common(int days, int Noperson) => days * Noperson * 1200;

    [LuxuryTax(10)]
    public double Senior(int days, int Noperson) => GetDiscount(days, Noperson) - 100;

    [LuxuryTax(10)]
    public double Woman(int days, int Noperson) => GetDiscount(days, Noperson) - 200;


    private double GetDiscount(int days, int Noperson)
    {
        double total = Common(days, Noperson);
        if ((days > 6) && (Noperson >= 4))
        {
            return total - 200 * days * Noperson;
        }
        else if (days > 6)
        {
            return total - 100 * days * Noperson;
        }
        else if (Noperson >= 4)
        {
            return total - 100 * days * Noperson;
        }
        return days * Noperson * 1200;
    }
}

File: Vishwas RNO 40/Java/Assign8/src/PremiumTours.java
package Tours;

public class PremiumTours {

    @LuxuryTax(10)
    public double common(int days, int noOfPersons) {

        return  days * noOfPersons * 1200;
    }

    @LuxuryTax(10)
    public double senior(int days, int noOfPersons){

        return getDiscount(days,noOfPersons) - 100;
    }

    @LuxuryTax(10)
    public double woman(int days, int noOfPersons){

        return getDiscount(days,noOfPersons) - 200;
    }

    private double getDiscount(int days, int noOfPersons) {
        
        double total = common(days, noOfPersons);
        if ((days > 6) && (noOfPersons >= 4))
        {
            return total - 200 * days * noOfPersons;
        }
        else if (days > 6)
        {
            return total - 100 * days * noOfPersons;
        }
        else if (noOfPersons >= 4)
        {
            return total - 100 * days * noOfPersons;
        }
        return days * noOfPersons * 1200;
    }
}